lib/toJsonKey.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+/**
+ * Helper for getting a reproducible, key-sorted string representation of object.
+ * Used to interpret an object as a falcor key.
+ * @function
+ * @param {Object} obj
+ * @return stringified object with sorted keys.
+ */
+function toJsonKey(obj) {
+    if (typeof obj !== "object" ||
+        obj === null ||
+        Object.prototype.toString.call(obj) !== '[object Object]') {
+        return JSON.stringify(obj);
+    }
+    return "{" +
+        Object.keys(obj).sort().reduce(function (str, key) {
+            if (obj[key] === undefined) {
+                return str;
+            }
+            return (str ? str + "," : str) +
+                toJsonKey(key) +
+                ":" +
+                toJsonKey(obj[key]);
+        }, "") +
+        "}";
+}
+
+function isJsonKey(key) {
+    if (typeof key !== 'string' ||
+        key.length <= 2 ||
+        key[0] !== '{' ||
+        key[key.length - 1] !== '}') {
+        return false;
+    }
```

Comment:
This is just a fast path so might as well just go with the simpler:
```js
    if (typeof key !== 'string' || key[0] !== '{') {
        return false;
    }
```

Reply:
done